### PR TITLE
Improve MongoKafka tests (GSI-988)

### PR DESCRIPTION
**Pull request by Cito**
* test MongoKafka with both str based and UUID based IDs (did that already for normal MongoDB provider)
* add tests for `find_one` and `find_all` when mapping values are UUIDs or datetimes (just to document that these need to be manually serialized when passed in the mapping)
* properly convert events with UUID and datetime values to JSON using a custom default serializer
* add some more tests for edge cases

### CODE REVIEW

File: tests/integration/test_mongodb.py [hunk 1]
```diff
@@ -441,13 +442,25 @@ async def test_dao_crud_happy(use_custom_id: bool, mongodb: MongoDbFixture):
 
     assert obtained_hits == {resource_updated, resource3}
 
+    # perform a search using values with non-standard data types
+    # (note that in this case we need to serialize these values manually):
+    mapping = {
+        id_field: str(getattr(resource, id_field)),
+        "field_d": resource.field_d.isoformat(),
+    }
```
Comment (TheByronHimes) [inline tests/integration/test_mongodb.py:450]: Can we apply the same SerDe behavior from `_default_json_serializer()` to mongodb query mappings before running the query? That would enable consistent behavior for read and write. Otherwise it's not intuitive that you can store an object with a datetime/uuid but have to search with the manually serialized values.

**Review by TheByronHimes — APPROVED** [review]

File: tests/integration/test_mongodb.py (hunk 1 above)
Comment (Cito) [reply tests/integration/test_mongodb.py]: Yes, that's not intuitive, that's why I at least wanted to document this in a test. But I think you're right, we should change this behavior, since otherwise switching to BSON would be a breaking change, because then manual conversion will not be necessary.

File: src/hexkit/providers/mongokafka/provider.py [hunk 2]
```diff
@@ -127,23 +130,23 @@ def get_delete_publish_func(
...
     """
 
-    async def publish_deletion(id_: str) -> None:
+    async def publish_deletion(id_: Any) -> None:
         """Publishes a deletion event and marks the deletion as published."""
         await event_publisher.publish(
             payload={},
             type_=DELETE_EVENT_TYPE,
-            key=id_,
+            key=str(id_),
```
Comment (TheByronHimes) [inline src/hexkit/providers/mongokafka/provider.py:143]: Why not use the `value_to_document` function here too?

**Review by TheByronHimes — CHANGES_REQUESTED** [review]
Generally looks good, just noticed a few things

File: src/hexkit/providers/mongokafka/provider.py [hunk 3]
```diff
@@ -221,7 +224,7 @@ def __init__(  # noqa: PLR0913
         self._publish_delete = publish_delete
         self._autopublish = autopublish
 
-    async def get_by_id(self, id_: Union[str, UUID]) -> Dto:
+    async def get_by_id(self, id_: Any) -> Dto:
```
[the remark this answers is not in the document]
Comment (Cito) [reply src/hexkit/providers/mongokafka/provider.py:226]: Good catch. Fixed in 8cf486.

File: src/hexkit/providers/mongokafka/provider.py [hunk 4]
```diff
@@ -351,6 +354,19 @@ async def find_all(self, *, mapping: Mapping[str, Any]) -> AsyncIterator[Dto]:
                 document, id_field=self._id_field, dto_model=self._dto_model
             )
 
+    def _convert_filter_values(self, value: Any) -> Any:
```
[the remark this answers is not in the document]
Comment (Cito) [reply src/hexkit/providers/mongokafka/provider.py:354]: Yes, we should probably do that. And so far we aren't testing this, will add that as well.

[the remark this answers is not in the document]
Comment (Cito) [reply src/hexkit/providers/mongokafka/provider.py:354]: Did that in 290c0ce.

Also added ID type and more rigorous testing.

**Review by TheByronHimes — APPROVED** [review]
Looks good to me, thank you!
